independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/ArcContainerImpl.java

```diff
@@ -630,22 +630,28 @@ private static Integer getAlternativePriority(InjectableBean<?> bean) {
                 String msg = "\n%1$s%1$s%1$s%1$s\n"
                         + "CDI: programmatic lookup problem detected\n"
                         + "-----------------------------------------\n"
-                        + "At least one bean matched the required type and qualifiers but was marked as unused and removed during build\n"
-                        + "Removed beans:\n\t- %2$s\n"
+                        + "At least one bean matched the required type and qualifiers but was marked as unused and removed during build\n\n"
+                        + "Stack trace element: %5$s\n"
                         + "Required type: %3$s\n"
                         + "Required qualifiers: %4$s\n"
+                        + "Removed beans:\n\t- %2$s\n"
                         + "Solutions:\n"
                         + "\t- Application developers can eliminate false positives via the @Unremovable annotation\n"
                         + "\t- Extensions can eliminate false positives via build items, e.g. using the UnremovableBeanBuildItem\n"
                         + "\t- See also https://quarkus.io/guides/cdi-reference#remove_unused_beans\n"
-                        + "\t- Enable the DEBUG log level to see the full stack trace to identify the method that performed the lookup\n"
+                        + "\t- Enable the DEBUG log level to see the full stack trace\n"
                         + "%1$s%1$s%1$s%1$s\n";
+                StackTraceElement[] stackTrace = Thread.currentThread().getStackTrace();
+                StackTraceElement element = Arrays.stream(stackTrace)
+                        .skip(1)
+                        .filter(ArcContainerImpl::isCallerElement)
+                        .findFirst().orElse(null);
```

Comment:
Hm... I think the Stackwalker API would be very useful here

Reply:
Ah, didn't know about this one. Interesting. Although for this particular use case we don't need anything special...

Reply:
Indeed, but I _think_ the StackWalker also perform better.

Reply:
Done.